packages/app/src/types.ts

```diff
@@ -0,0 +1,145 @@
+import { FirebaseModule } from './internal';
+import FirebaseApp from './FirebaseApp';
+
+export type DeleteAppType = () => Promise<never | void>;
+
+export interface FirebaseAppImpl {
+  /**
+   * The name (identifier) for this App. '[DEFAULT]' is the default App.
+   */
+  readonly name: string;
+  readonly options: FirebaseOptionsImpl;
+  readonly automaticDataCollectionEnabled: boolean;
+
+  /**
+   * The (read-only) configuration options from the app initialization.
+   */
+
+  extendApp(extendedProps: any): void;
+  delete(): Promise<void>;
+  toString(): string;
+  SDK_VERSION: string;
+  apps: FirebaseAppImpl[];
+  app(name?: string): Promise<FirebaseAppImpl>;
+  initializeApp(
+    options: FirebaseOptionsImpl,
+    configOrName: FirebaseConfigImpl | string,
+  ): Promise<FirebaseAppImpl>;
+
+  /**
+   * Make this app unusable and free up resources.
+   */
+  // eslint-disable-next-line @typescript-eslint/adjacent-overload-signatures
+  delete(): Promise<void>;
+
+  utils(): any;
+}
+
+export interface FirebaseModuleNamespaceImpl {
+  statics: any; //this prop will have to be removed & extended for each namespace as each package has its own statics type
+  version: string;
+  namespace: NamespaceTypes;
+  nativeModuleName: any; //this prop will have to be removed & extended for each namespace as each package has its own statics type
+  nativeEvents: any; //this prop will have to be removed & extended for each namespace as each package has its own statics type
```

Comment:
On the previous types we had https://github.com/invertase/react-native-firebase/blob/master/packages/app/lib/index.d.ts#L234 which allowed custom types and statics e.g. auth does: https://github.com/invertase/react-native-firebase/blob/master/packages/auth/lib/index.d.ts#L1690

Reply:
Just clarify, it is ok to delete this `def`?